src/fidasim.f90

```diff
@@ -6117,8 +6217,28 @@ subroutine gyro_step(vi, fields, r_gyro)
         else
             rg_uvw = matmul(beam_grid%basis,r_gyro)
         endif
-        rg_r = rg_uvw(1)*cos(phi) + rg_uvw(2)*sin(phi)
-        r_gyro = r_gyro*(1 - rg_r/(2*R))
+        rg_r = rg_uvw(1) * cos(phi) + rg_uvw(2) * sin(phi)
+        b_rtz(1) = fields%br / fields%b_abs
+        b_rtz(2) = fields%bt / fields%b_abs
+        b_rtz(3) = fields%bz / fields%b_abs
+        cuvrxb(1) = -fields%dbt_dz / fields%b_abs
+        cuvrxb(2) = (fields%dbr_dz - fields%dbz_dr) / fields%b_abs
+        cuvrxb(3) = fields%dbt_dr / fields%b_abs
+        term1 = vpar * one_over_omega * dot_product(b_rtz,cuvrxb)
+        grad_B(1) = (fields%br*fields%dbr_dr + fields%dbt_dr+fields%bz*fields%dbz_dr) /&
+                    fields%b_abs
+        grad_B(2) = 0.0
+        grad_B(3) = (fields%br * fields%dbz_dr + fields%dbt_dz + fields%bz*fields%dbz_dz) /&
+                    fields%b_abs
+        rg_rtz(1) = rg_uvw(1) * cos(phi) + rg_uvw(2) * sin(phi)
+        rg_rtz(2) = 0.0
+        rg_rtz(3) = rg_uvw(3)
+        term2 = -1.0 / (2.0 * fields%b_abs) * dot_product(rg_rtz,grad_B)
+        r_gyro = r_gyro * (1.0 - term1 - term2)
+        if(1.0 - term1 - term2 .le. 0.0) then
+            write(*,'(a)') '---- Error in subroutine gyro_step ----'
```

Comment:
Could you change the error message to have the same format as the other messages in the code.